index.js

```diff
@@ -68,6 +68,15 @@ module.exports = {
   included: function() {
     this._super.included.apply(this, arguments);
 
+    let isApp = !this.project.isEmberCLIAddon();
+
+    let hasEmberFetch = this.project.addons.map(addon => addon.name).includes('ember-fetch');
+
+    if(isApp && !hasEmberFetch) {
+      throw new Error(`Ember fetch is not installed as top-level dependency of the application. Add ember-fetch as dependecy in application's package.json.
```

Comment:
@rwjblue should we start as a warning, and do a major version release that turns this into an error, or...

Reply:
Yea I think we should follow deprecation workflow, error in next major (7.0.0).

Reply:
I agree with that. what is the general flow that we follow for deprecation? What is the release branch we can update with warning?

Reply:
We chatted about this. Using ember-fetch in a fastboot project will error anyway so this is not a deprecation, but rather making error more explicit.